code/script.js

```diff
@@ -1,325 +1,824 @@
   // This variable stores what option group (category) the question belongs to.
   // We also need a variable that stores the actual value of the question we've selected.
-  // const value =
+  const value = questions.options[questions.selectedIndex].text;
 
   currentQuestion = {
     category: category,
-    // value: value
-  }
-}
+    value: value,
+  };
+
+  // counter will be added one when this function is called.
+  counter++;
+  counterDisplay.innerText = counter;
+
+  // After a user select an option and click filterBtn, this function will call to check what will happen the next
+  checkQuestion();
+};
 
 // This function should be invoked when you click on 'Find Out' button.
 const checkQuestion = () => {
-  const { category, value } = currentQuestion
+  const { category, value } = currentQuestion;
+
+  // Getting rid of white space of a value(green eye-> greeneye/ "wedlock" needs to be concated due to a value from option)
+  valueNoWhiteSpace = category === "wedlock" ? value + category : value.replace(/\s/g, "");
+
+  /*****************************************************************************************/
+  // Compare the currentQuestion details with the secret person details in a different manner based on category
+  if (category === "hair" || category === "eyes" || category === "wedlock") {
+    if (`${secret[category]}${category}` === valueNoWhiteSpace) {
+      filterCharacters(true);
+    } else {
+      filterCharacters(false);
+    }
+    /*****************************************************************************************/
+  } else if (category === "pet" || category === "accessories" || category === "other") {
+    // These categories are made from arrays. So it needs to be carefully validated.
+
+    // to make it easier to identigy key in object for step 5. Input value and proparty in object are different, so I am fixing it here.
+
+    switch (valueNoWhiteSpace) {
+      case "ahat":
+        valueAsKey = "hat";
+        break;
+      case "glasses":
+        valueAsKey = "glasses";
+        break;
+      case "asmokinghabit":
+        valueAsKey = "smoker";
+        break;
+      case "acat":
+        valueAsKey = "cat";
+        break;
+      case "adog":
+        valueAsKey = "dog";
+        break;
+      case "arabbit":
+        valueAsKey = "rabbit";
+        break;
+    }
 
-  // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
-  // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
-  if (category === 'hair' || category === 'eyes') {
+    const secretCategory = secret[category];
+    const secretCategoryString = secretCategory.toString();
 
-  } else if (category === 'accessories' || category === 'other') {
+    // There are some varieties with these arrays. But other option has only two options. Smoker or not. So I validate it separetly.
```

Comment:
I didn't really get what you wrote here.. But it seems to be working! Great job figuring this out! :)

Reply:
I did so because I picked text from select instead of value. It was a really bad idea as you can see. I thought of changing it, but I spent a lot of time to solve this issue, so I decided to keep it.